### PR TITLE
WIP: Implementation for temperature of full run (year) provided externally

### PR DESCRIPTION
**Pull request by mbechtold**
zero diff with the current main, only affects external calls from LIS.

### CODE REVIEW

File: src/global.f90 [hunk 1]
```diff
@@ -16109,6 +16118,74 @@ subroutine SetTactWeedInfested(TactWeedInfested_in)
...
+    integer(int32), intent(in) :: i
+    real(dp) :: TminRun_i
+
+    TminRun_i = TminRun(i)
+end function GetTminRun_i
+
+
+subroutine SetTminRun(TminRun_in)
+    !! Setter for the "TminRun" global variable.
+    real(dp), dimension(365), intent(in) :: TminRun_in
```
Comment (lbusschaert) [inline src/global.f90:16140]: what if leap year?

File: src/tempprocessing.f90 [hunk 2]
```diff
@@ -894,6 +903,32 @@ integer(int32) function GrowingDegreeDays(ValPeriod, FirstDayPeriod, Tbase, &
...
+            TDayMin_local = GetTminRun_i(i)
+            TDayMax_local = GetTmaxRun_i(i)
+            DayGDD = DegreesDay(Tbase, Tupper, TDayMin_local, &
+                                        TDayMax_local, &
+                                        GetSimulParam_GDDMethod())
+            GDDays = GDDays + DayGDD
+            RemainingDays = RemainingDays - 1
+
+            do while ((RemainingDays > 0) &
+                        .and. (i<365))
```
Comment (lbusschaert) [inline src/tempprocessing.f90:918]: Same here

File: src/tempprocessing.f90 [hunk 3]
```diff
@@ -894,6 +903,32 @@ integer(int32) function GrowingDegreeDays(ValPeriod, FirstDayPeriod, Tbase, &
...
+                        TDayMax_local = GetTmaxRun_i(i)
+                        DayGDD = DegreesDay(Tbase, Tupper, TDayMin_local, &
+                                            TDayMax_local, &
+                                            GetSimulParam_GDDMethod())
+                        GDDays = GDDays + DayGDD
+                        RemainingDays = RemainingDays - 1
+            end do
+
+           if (RemainingDays > 0) then
+                 GDDays = undef_int
```
Comment (lbusschaert) [inline src/tempprocessing.f90:930]: What would that mean? That the cropping period is longer than the simulation period? 

Related to this: when running in GDD, we'll have to be careful in regions where we don't have enough GDDays to complete the crop cycle in one year. I guess we'll just stop the growth and restart a new simulation period, but something to keep in mind when we finally do some science with the code.

**Review by lbusschaert — APPROVED** [review]
Looks good! Just wondering about leap year and another small thing (see comments in the code)

File: src/tempprocessing.f90 (hunk 3 above)
Comment (mbechtold) [reply src/tempprocessing.f90:930]: Yes, we will just stop the growth and restart a new simulation period. 
Should we here include the rewinding and set i back to 1 when GetSimulation_ToDayNr()-GetSimulation_FromDayNr() is reached to catch those cases?

**Pull request merged**